### PR TITLE
Add error handling

### PR DESCRIPTION
**Pull request by neehar18**
In reference to: https://github.com/redhat-developer/vscode-granite/issues/5

### CODE REVIEW

File: webviews/src/App.tsx [hunk 1]
```diff
@@ -109,6 +109,15 @@ function App() {
...
       }
+      case 'error': {
+        // Generic error handling: capture the error from the backend and display it to the user
+        const errorMessage = payload.data.error;
+        alert(`An error occurred: ${errorMessage}`);
+
+        // Re-enable the UI so the user can try again
+        setEnabled(true);
+        break;
+      }
```
Comment (fbricon) [inline webviews/src/App.tsx:120]: this block is never reached

**Comment by neehar18** [comment]
Here are my test results from mockServer.ts:

* For the case when there are errors during the pull:

Screenshot 2024-10-01 at 9 21 47 PM


* For the case when there is a mismatch in the configuration:

Screenshot 2024-10-01 at 9 27 33 PM

File: src/ollama/mockServer.ts [hunk 2]
```diff
@@ -183,6 +182,12 @@ export class MockServer extends OllamaServer implements IModelServer {
           total: layer.size
         });
 
+        // Throw error if model is "granite-code:34b" and we reach 10% progress
+        // if (modelName === "granite-code:34b" && layer.progress / layer.size >= 0.1) {
+        //  reject(new Error('Simulated error: Insufficient space while pulling granite-code:34b.'));
+        //  return;
+        // }
```
Comment (fbricon) [inline src/ollama/mockServer.ts:189]: please uncomment

File: src/ollama/mockServer.ts [hunk 3]
```diff
@@ -209,7 +215,10 @@ export class MockServer extends OllamaServer implements IModelServer {
     tabModelName: string | null,
     embeddingsModelName: string | null
   ): Promise<void> {
-    //TODO, if chatModelName=granite-code:3b and tabModelName=granite-code:20b, throw an exception
+    // Throw an error if conflicting models are selected
+    // if (chatModelName === "granite-code:3b" && tabModelName === "granite-code:20b") {
+    //   throw new Error('Simulated error: Conflicting models selected for chat and tab completion.');
+    // }
```
Comment (fbricon) [inline src/ollama/mockServer.ts:221]: please uncomment

File: src/panels/setupGranitePage.ts [hunk 4]
```diff
@@ -355,19 +355,22 @@ export class SetupGranitePage {
         embeddingsModelId: graniteConfiguration.embeddingsModelId ?? 'none',
       });
     } catch (error: any) {
-      //if error is CancellationError, then we can ignore it
-      if (error instanceof CancellationError || error?.name === "Canceled") {
-        return;
-      }
```
Comment (fbricon) [inline src/panels/setupGranitePage.ts:361]: No you need to keep it or else you'll get an error popup. User cancelled on purpose, so no popup is necessary IMO.
Screenshot 2024-10-01 at 18 29 46

**Review by fbricon — CHANGES_REQUESTED** [review]

**Review by fbricon — APPROVED** [review]

**Comment by fbricon** [comment]
Thanks @neehar18

**Pull request merged**